### ISSUE TITLE
`melt` panic on degenerate case

### ISSUE THREAD

**Issue by gab23r** [pola-rs/polars#10075]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python

import polars as pl
df = pl.DataFrame({'only_one_column': [1,2,3]})

melted = df.melt('only_one_column')
# PanicException: called `Option::unwrap()` on a `None` value
```

### Issue description

This is a degenerate case where there is no column to melt. I think we should not panic here. 
Pandas' behavior seems good to me; it returns an empty frame.

### Expected behavior
```python
assert melted.to_dict(False) == {'only_one_column': [], 'variable': [], 'value': []}
assert melted.schema == {'only_one_column': pl.Int64, 'variable': pl.Utf8, 'value': pl.UInt32}
```
Basically I expect the same thing as if I was doing:

```python

melted = (
    df
    .with_row_count()
    .melt('only_one_column')
    .filter(pl.col('variable') != 'row_nr')
)

```

### Installed versions

<details>

```
Replace this line with the output of pl.show_versions(). Leave the backticks in place.
```

</details>

**Comment by cmdlineluser** [comment]
https://github.com/pola-rs/polars/blob/6befc52a46bf4fa1777584bf5e673cee880af5f5/polars/polars-core/src/frame/explode.rs#L291

It also happens elsewhere for a `LazyFrame`:
```python
thread '<unnamed>' panicked at 'called `Option::unwrap()` on a `None` value', 
/Users/user/git/polars/polars/polars-lazy/polars-plan/src/logical_plan/builder_functions.rs:60:43
```
https://github.com/pola-rs/polars/blob/6befc52a46bf4fa1777584bf5e673cee880af5f5/polars/polars-lazy/polars-plan/src/logical_plan/builder_functions.rs#L60

Probably makes sense to address both at the same time.

**Comment by stinodego** [comment]
Given that `melt` has been included mostly as a convenience for pandas users, we should follow pandas behavior here and return an empty frame.
